services/cannon/src/Cannon/API/Internal.hs

```diff
@@ -16,71 +18,106 @@
 -- with this program. If not, see <https://www.gnu.org/licenses/>.
 
 module Cannon.API.Internal
-  ( sitemap,
+  ( InternalAPI,
+    internalServer,
   )
 where
 
 import Cannon.App
 import qualified Cannon.Dict as D
 import Cannon.Types
 import Cannon.WS
+import Conduit
 import Control.Monad.Catch
 import Data.Aeson (encode)
+import qualified Data.ByteString as S
 import qualified Data.ByteString.Lazy as L
-import Data.Id (ConnId, UserId)
-import Data.Swagger.Build.Api hiding (Response)
+import Data.Conduit.List
+import Data.Id hiding (client)
 import Gundeck.Types
 import Gundeck.Types.BulkPush
-import Imports hiding (head)
-import Network.HTTP.Types
-import Network.Wai
-import Network.Wai.Predicate
-import Network.Wai.Routing
-import Network.Wai.Utilities
+import Imports
+import Servant
+import Servant.Conduit ()
 import System.Logger.Class (msg, val)
 import qualified System.Logger.Class as LC
+import Wire.API.ErrorDescription
+import Wire.API.Routes.MultiVerb
+import Wire.API.Routes.Named
+import Wire.API.Routes.NoContentWithStatusCodeVerb
 
-sitemap :: Routes ApiBuilder Cannon ()
-sitemap = do
-  get "/i/status" (continue (const $ return empty)) true
-  head "/i/status" (continue (const $ return empty)) true
-
-  post "/i/push/:user/:conn" (continue pushH) $
-    capture "user" .&. capture "conn" .&. request
-
-  post "/i/bulkpush" (continue bulkpushH) $
-    request
+newtype PushNotificationStream = PushNotificationStream
+  { getPushNotificationStream :: ConduitT () ByteString (ResourceT WS) ()
+  }
+  deriving newtype (FromSourceIO ByteString)
 
-  head "/i/presences/:uid/:conn" (continue checkPresenceH) $
-    param "uid" .&. param "conn"
+type InternalAPI =
+  "i"
+    :> ( Named
+           "get-status"
+           ( "status"
+               :> NoContentWithStatusCodeVerb 'GET 200
+           )
+           :<|> Named
+                  "head-status"
+                  ( "status"
+                      :> NoContentWithStatusCodeVerb 'HEAD 200
+                  )
+           :<|> Named
+                  "push-notification"
+                  ( "push"
+                      :> Capture "user" UserId
+                      :> Capture "conn" ConnId
+                      :> StreamBody NoFraming OctetStream (SourceIO ByteString)
```

Comment:
Did you mean to use `PushNotificationStream` here instead of `SourceIO ByteString`? Otherwise, what is the point of that `newtype`?